odoo/addons/base/models/ir_model.py

```diff
@@ -1980,8 +1981,25 @@ def delete(records):
         # remove models
         delete(self.env['ir.model'].browse(model_ids))
 
+        # attempt to remove undeletable model data that may have become deletable again because
+        # of the tables being dropped just above
+        for data in self.browse(undeletable_ids).exists():
+            record = self.env[data.model].browse(data.res_id)
+            try:
+                with self.env.cr.savepoint():
+                    if record.exists():
+                        # record exists therefore the data is still undeletable,
+                        # remove it from module_data
+                        module_data -= data
+                        continue
+            except psycopg2.ProgrammingError:
+                # This most likely means that the record does not exist, in SQL this could be
+                # rougly equivalent to `SELECT id FROM table WHERE id=record.id` and it may raise
+                # a ProgrammingError because the table no longer exists (and thus nor does the
+                # record), also applies to ir.model.fields, constraints, etc.
+                ...
```

Comment:
```suggestion
                # This most likely means that the record does not exist, since record.exists()
                # is rougly equivalent to `SELECT id FROM table WHERE id=record.id` and it may raise
                # a ProgrammingError because the table no longer exists (and so does the
                # record), also applies to ir.model.fields, constraints, etc.
                pass
```